graph.py

```diff
@@ -231,94 +243,112 @@ def __init__(self, directed: bool, n: int=0, simple: bool=False, name: str='G'):
+
+        :return: A textual representation of the vertices and edges of this graph.
         """
-        A user-friendly representation of this graph
-        :return: A textual representation of the vertices and edges of this graph
-        """
-        return "G" + self._name + ':\nV=[' + ", ".join(map(str, self._v)) + ']\nE=[' + ", ".join(map(str, self._e)) + ']'
+
+        return self._name + ':\n' \
+                            'V=[' + ", ".join(map(str, self._v)) + ']\n' \
+                                                                   'E=[' + ", ".join(map(str, self._e)) + ']'
 
     def _next_label(self) -> str:
+        """Generate a unique label for vertices in the graph.
+
+        :return: A unique label.
         """
-        Generates unique labels for vertices within the graph
-        :return: A unique label
-        """
+
         result = self._next_label_value
         self._next_label_value += 1
         return str(self._name) + '_' + str(result)
 
     @property
     def name(self) -> str:
-        """
-        Name of the graph
+        """Name of the graph.
+
         :return: Name of the graph
         """
+
         return self._name
 
+    @name.setter
+    def name(self, name: str):
+        self._name = name
+
     @property
     def simple(self) -> bool:
+        """Whether this graph is a simple graph, i.e., it does not have multi-edges or loops.
+
+        :return: Whether this graph is simple.
         """
-        Whether the graph is a simple graph, that is, it does not have multi-edges or loops.
-        :return: Whether the graph is simple
-        """
+
         return self._simple
 
     @property
     def directed(self) -> bool:
+        """Whether this graph behaves as a directed graph.
+
+        :return: Whether the graph is directed.
         """
-        Whether the graph behaves as a directed graph
-        :return: Whether the graph is directed
-        """
+
         return self._directed
 
     @property
     def vertices(self) -> List["Vertex"]:
-        """
-        :return: The `set` of vertices of the graph
-        """
+        """:return: The `set` of vertices of this graph."""
+
         return list(self._v)
 
     @property
     def edges(self) -> List["Edge"]:
-        """
-        :return: The `set` of edges of the graph
-        """
+        """:return: The `set` of edges of this graph."""
+
         return list(self._e)
 
+    @property
+    def order(self) -> int:
+        """The number of vertices of this graph."""
+        return len(self.vertices)
+
+    @property
+    def size(self) -> int:
+        """The number of edges of this graph."""
+
+        return len(self.edges)
+
     def __iter__(self):
-        """
-        :return: Returns an iterator for the vertices of the graph
-        """
-        return iter(self._v)
+        """:return: Returns an iterator for the vertices of the graph."""
 
-    def __len__(self) -> int:
```

Comment:
Hebben we nu geen __len__ functie meer?

Reply:
Nee, nu moet je dus `order` gebruiken

Reply:
Nee, want aldus https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)#Graph heeft een graaf een size en een order. Wat is dan de lengte van de graaf? Slechte naamgeving, dat is het. Dus weg ermee en gewoon nette methoden voor size en order.